### DISCUSSION TITLE
Most efficient way to monitor volume of AudioStream?

### DISCUSSION THREAD

**Discussion by leelx78** [pschatzmann/arduino-audio-tools#858] [Q&A] [answered]
Hi @pschatzmann what's the recommended way to monitor the volume of an AudioStream (PCM)? I tried VolumeStream but that seems to be mostly for setting the volume rather than monitoring its output? 

Something like a moving average of the signal would work great so that we can use that to detect if there is actual input or noise in the channel, with a time constant.

If I need to write an extension of AudioStream with this feature it's no problem, I just wanted to check with you first before reinventing the wheel :D 

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
The class you are looking for is call VolumeOutput!
